client/src/components/itemColorOptions.jsx

```diff
@@ -1,23 +1,41 @@
-import React from 'react';
+import React, { useState } from 'react';
 import ColorOption from './colorOption.jsx';
 import { SectionDescriptionText } from '../style.js';
 import { ColorBoxBlack, ColorBoxGray, ColorBoxWhite } from '../style.js';
 
 
-const ItemColorOptions = (props) => (
-  //map color options to colorOption component
-  //format 3 squares
-  <div>
-    <SectionDescriptionText>Color:</SectionDescriptionText> <b>{props.color}</b>
-    <br></br>
-    <ColorBoxBlack></ColorBoxBlack>
-    <ColorBoxWhite></ColorBoxWhite>
-    <ColorBoxGray></ColorBoxGray>
+const ItemColorOptions = (props) => {
+
+  const [color, setColor] = useState(props.color);
+
+  //handle box clicks which will take in productId and return productId
+
+  //pass productId up to app.jsx
+  return (
     <div>
-      {props.similarItems.map(item => <ColorOption item={item} />
-      )}
+      <SectionDescriptionText>Color:</SectionDescriptionText> <b>{color}</b>
+      <br></br>
+      <ColorBoxBlack onClick={() => props.handleColorBoxClick(props.similarItems[0])} onMouseEnter={() => setColor('Black')} onMouseLeave={() => setColor(props.color)}></ColorBoxBlack>
+      <ColorBoxWhite onClick={() => props.handleColorBoxClick(props.similarItems[1])} onMouseEnter={() => setColor('White')} onMouseLeave={() => setColor(props.color)}></ColorBoxWhite>
+      <ColorBoxGray onClick={() => props.handleColorBoxClick(props.similarItems[2])} onMouseEnter={() => setColor('Gray')} onMouseLeave={() => setColor(props.color)}></ColorBoxGray>
+      <div>
+        {props.similarItems.map(item => <ColorOption item={item} />
+        )}
+      </div>
     </div>
-  </div>
-);
+
+  );
+};
+
+
+
+//need to handle hover over color box'
+//triggers the currentColor to change
+
+//need to hanlde click on color box
+//triggers the whole state of the app to change a new item
+//returns a productId
+//pass this object up to the top container (app)
```

Comment:
I think you've done most or all of this. Maybe you can consider deleting or editing this code

Reply:
I will go ahead and do that.